library/core/src/intrinsics.rs

```diff
@@ -1748,7 +1748,8 @@ extern "rust-intrinsic" {
 /// Checks whether `ptr` is properly aligned with respect to
 /// `align_of::<T>()`.
 pub(crate) fn is_aligned_and_not_null<T>(ptr: *const T) -> bool {
-    !ptr.is_null() && ptr as usize % mem::align_of::<T>() == 0
+    use crate::ptr;
+    ptr_guaranteed_ne(ptr, ptr::null()) && ptr.align_offset(mem::align_of::<T>()) == 0
```

Comment:
Please leave the code of this function unchanged compared to what it was before your PR.

Reply:
Ok :)